### ISSUE TITLE
0.47.0 live reload source revert after press ESC

### ISSUE THREAD

**Issue by harrisyu** [nesbox/TIC-80#346]
![live-reload problem](https://user-images.githubusercontent.com/4844464/31826378-770d106c-b5e7-11e7-8f3a-8d741089db75.gif)
Code will auto load after save in developer mode, but revert after press ESC.

**Comment by matheuslessarodrigues** [comment]
@nesbox I'm trying to reason about this bug but can't figure out what I need to look for. Is there a place where TIC reloads/resets code when ESC is pressed? When I reload code, I only change the `char* studio.code.data` (studio.c:1755) buffer. Is there another place I should also reload code to? Thanks!

**Comment by nesbox** [comment]
@matheuslessarodrigues no, there is one place for the code buffer
It's strange, I will try to investigate it too...

**Comment by matheuslessarodrigues** [comment]
@nesbox Maybe it has something to do with this `embed` structure in `console.c`. There's this line in `console.c:2318`:
```c
bool loaded = loadFileIntoBuffer(console, embed.file.code.data, name);
```

So, when injecting code (the first time), it loads it into this `embed.file.code.data` which is different from `studio.code.data` in `studio.c:1755`.

How is this `embed` struct used when injecting code? Do you remember if it's used later in the code (like when returning to code editor by pressing ESC)?

**Comment by nesbox** [comment]
Seems I found a problem, fixed it here c5cebbc
closing
